src/commands/test/contract_fork.py

```diff
@@ -0,0 +1,28 @@
+import copy
+from typing import Optional
+
+from starkware.starknet.testing.starknet import Starknet
+from starkware.starknet.testing.contract import StarknetContract
+from starkware.starknet.definitions.general_config import StarknetGeneralConfig
+from starkware.starknet.testing.state import StarknetState
+
+
+class ForkableStarknet(Starknet):
+    @classmethod
+    async def empty(
+        cls, general_config: Optional[StarknetGeneralConfig] = None
+    ) -> "Starknet":
+        return ForkableStarknet(
+            state=await StarknetState.empty(general_config=general_config)
+        )
+
+    def plug_from_different_state(self, deployed_contact: StarknetContract):
```

Comment:
```suggestion
    def copy_and_adapt_contract(self, starknet_contract: StarknetContract):
```
- [ ] fix typo: `contact` -> `contract`
- Does the contract have to be deployed?
- The name of this function is a bit confusing. I'd change it to `copy_and_adapt_contract`.

Reply:
Yea, name is terrible, your is much better, thanks. Contact doesn't have to be deployed, we copy state